torchtext/data/functional.py

```diff
@@ -61,13 +52,7 @@ def load_sp_model(spm_path):
         >>> from torchtext.data.functional import load_sp_model
         >>> sp_model = load_sp_model("m_user.model")
     """
-    try:
-        import sentencepiece as spm
-    except ModuleNotFoundError:
-        raise ImportWarning("Please install sentencepiece")
-    sp_model = spm.SentencePieceProcessor()
-    sp_model.Load(spm_path)
-    return sp_model
+    return torch.ops.torchtext.load_sp_model(spm_path)
 
 
 def sentencepiece_numericalizer(sp_model):
```

Comment:
This function is the tokenizer, which tokenizes a list of text strings to a list of index with a pretrained sp model. We should add the cpp extension for this one.

Reply:
Offline discussion: Handling generator is a more complicated, I will work on it on the next PR.